concepts/pointers/about.md

```diff
@@ -0,0 +1,404 @@
+    int my_ints[] = {1, 42};
+    int *my_ptr = my_ints;
+    
+    // prints: my_ints[0] is 1. *++my_ptr is 42.
+    printf("my_ints[0] is %u. *++my_ptr is %u.\n", my_ints[0], *++my_ptr);
+    
+    // prints: my_ints[1] is 43. ++*my_ptr++ is 43.
+    printf("my_ints[1] is %u. ++*my_ptr++ is %u.\n", my_ints[1], ++*my_ptr);
+
+    return 0;
+}
+```
+
+Both the `*` and `++` have the same precedence, but they evaluate from right to left.
+In the first line `my_ptr` is first incremented to the second element, which is then dereferenced.
+In the second line `my_ptr`, which was incremented to the second element in the first line, is first dereferenced, and then the dereferenced value is incremented.
+If we add this line to the above example
+
+```c
+// prints: my_ints[1] is 43. *++my_ptr is (some garbage value).
+printf("my_ints[1] is %u. *++my_ptr is %u.\n", my_ints[1], *++my_ptr);
+```
+
+In that line we have incremented the pointer to be beyond the array.
+We have overrun the array and we are pointing at the next four bytes in memory which the pointer interprets as an int, but, in fact, those bytes could belong to another variable or variables.
+Those four bytes could  be part of a `double` or they could be four `char`s.
+The problem is we shouldn't even be looking at them with our pointer.
+
+We could have used
+
+```c
+// prints: my_ints[2] is (some garbage value).
+printf("my_ints[2] is %u", my_ints[2]);
+```
+
+Since array indexing is implemented with a pointer, the same danger of overunning the array applies.
+
+And, since a string is an array of `char`s, pointers can be used with strings, too.
+A string is often declared using pointer syntax, like so
+
+```c
+char *my_string = "This is my string";
+// instead of
+// char my_string[] = "This is my string";
+```
+
+One question, though.
+Why do we put the star next to the identifier?
+Why not put it next to the type, since it's a pointer to that type?
+Because this
+
+```c
+int* my_ptr1, my_ptr2_that_is_really_an_int;
+```
+
+does not declare two pointers to an int, but declares a pointer to an int, and an int.
+To declare two pointers to an int would be like so
+
+```c
+int *my_ptr1, *my_ptr2;
+```
+
+So it's good to put the `*` near the identifier to minimize ambiguity.
+Or declare only one identifier per line.
+And, if you haven't had enough of `*`, welcome to...
+
+## Pointer to Pointers
+
+It is common to see arrays of strings represented like so
+
+```c
+#include <stdio.h>
+
+void print_strings(char **passed_in_strings) {
+    printf("%s\n", *passed_in_strings);
+    printf("%s\n", *(passed_in_strings + 1));
+}
+
+int main() {
+    char *my_strings[] = {"One string", "Two string"};
+    print_strings(my_strings);
+    return 0;
+}
+```
+
+It may not be obvious, but the array of strings is declared and initialized as a pointer to an array of `char` arrays.
+When the array of `char` arrays is passed into `print_strings`, the signature describes it as a pointer to pointer(s) to `char`.
```

Comment:
```suggestion
When the array of `char` arrays is passed into `print_strings()`, the signature describes it as a pointer to pointer(s) to `char`.
```

Typically, when referring to a C function, people include the `()`.
I think this is only widely held convention rather than any rule we have though.
Perhaps @ryanplusplus could opine on whether it matters or not.

Reply:
I'll commit it for now.